src/ipcache.cc

```diff
@@ -280,7 +280,7 @@ IpCacheLookupForwarder::forwardLookup(const char *error)
     // are sequential. Give it just the new, yet-unaccounted-for delay.
     if (receiverObj.set()) {
         if (auto receiver = receiverObj.valid()) {
-            receiver->noteLookup(Dns::LookupDetails(error, additionalLookupDelay()));
+            receiver->noteLookup(Dns::LookupDetails(SBuf(error), additionalLookupDelay()));
```

Comment:
In order to use std::optional this caller code needs to determine whether an error message exists and construct LookupDetails with different constructors - one which sets error, and one which only sets wait time (not yet existing).